lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/fs.rb

```diff
@@ -711,7 +711,11 @@ def cmd_ls(*args)
         cmd_ls_help
         return 0
       when nil
-        path = val
+        if client.platform == 'windows' && val.include?('%')
+          path = client.fs.file.expand_path(val)
+        else
+          path = val
+        end
```

Comment:
I talked this over with the team and they mentioned that this should actually use the `PATH_REGEX` to be consistent with the other uses in this file.

That would make the whole thing:
```suggestion
        path = val
        path = client.fs.file.expand_path(path) if path =~ PATH_EXPAND_REGEX
```

Sorry for the confusion. Once this has been changed over, I'll test this once more and land it. Thanks!

Reply:
Thank you. I've revised it as you said.